docker/transformers-past-gpu/Dockerfile

```diff
@@ -1,4 +1,4 @@
-ARG BASE_DOCKER_IMAGE="nvidia/cuda:11.2.2-cudnn8-devel-ubuntu20.04"
+ARG BASE_DOCKER_IMAGE
```

Comment:
What does this change do?

Reply:
This is for past CI - different versions of `torch` and `tensorflow` CI require different base docker images.
It's doesn't make a lot of sense to define a default `BASE_DOCKER_IMAGE` in the past CI docker file.


I move them to `utils/past_ci_versions.py`. This information will be fetched in `.github/workflows/build-past-ci-docker-images.yml`, and pass to the docker files via
```
build-args: |
  REF=main
  BASE_DOCKER_IMAGE=${{ steps.get-base-image.outputs.base_image }}
```